spswarehouse/powerschool/powerschool_calpads.py

```diff
@@ -0,0 +1,312 @@
+import time
+from spswarehouse.powerschool.powerschool import PowerSchool
+
+class PowerSchoolCALPADS(PowerSchool):
+    """
+    This class extends the PowerSchool class in order to download CALPADS
+    reports.
+    """
+    
+    def __init__(self, username: str=None, password: str=None, host: str=None, headless: bool=True, download_location: str='.'):
+        super().__init__(username, password, host, headless, download_location)
+
+    def download_calpads_report_for_school(self, school_full_name: str, submission_window: str, report_name: str, 
+            school_subdistrict_name: str, file_postfix: str, destination_directory_path: str, 
+            report_parameters: dict, validation_only_run: bool=False):
+        """
+        Switches to the desired school in PowerSchool and calls the function to generate the desired
```

Comment:
1. Should probably add a few details on what kind of value the various arguments here take. In particular `report_name`, and `school_subdistrict_name` could be opaque.
2. I wonder if the various `report_name ==` checks should also check for just the abbreviation? In particular, I think most people that do CALPADS don't refer to the report by actual name, but by abbreviation. The CALPADS class also refers to file types exclusively by abbreviation.

Reply:
Ah, I didn't follow the code that far in.

I lean coding it into the package, since the text is the same for all PowerSchool users, but it does make the whole thing less flexible.